File: src-tauri/src/commands/prompts.rs
//! Prompt (CLAUDE.md) preset commands.

use tauri::State;

use crate::database::dao::{self, PromptRenameScope};
use crate::error::AppResult;
use crate::prompts::{self, LivePrompt, PromptDetail, PromptInfo, PromptTarget};
use crate::store::AppState;

/// List all stored presets (names + mtimes, no content).
#[tauri::command]
pub fn list_prompts(target: Option<PromptTarget>) -> AppResult<Vec<PromptInfo>> {
    prompts::list_prompts(target.unwrap_or_default())
}

/// Read one preset's content for the editor.
#[tauri::command]
pub fn read_prompt(name: String, target: Option<PromptTarget>) -> AppResult<PromptDetail> {
    prompts::read_prompt(target.unwrap_or_default(), &name)
}

/// Create or overwrite a preset.
#[tauri::command]
pub fn save_prompt(name: String, content: String, target: Option<PromptTarget>) -> AppResult<()> {
    prompts::save_prompt(target.unwrap_or_default(), &name, &content)
}

/// Rename a preset and cascade profile `prompt_id` references.
#[tauri::command]
pub fn rename_prompt(
    old_name: String,
    new_name: String,
    target: Option<PromptTarget>,
    state: State<'_, AppState>,
) -> AppResult<()> {
    let target = target.unwrap_or_default();
    prompts::rename_prompt(target, &old_name, &new_name)?;
    // Profiles 只引用 claude/codex 作用域的 prompt；opencode 预设无需级联。
    let scope = match target {
        PromptTarget::ClaudeCode => Some(PromptRenameScope::ClaudeCode),
        PromptTarget::Codex => Some(PromptRenameScope::Codex),
        PromptTarget::OpenCode => None,
    };
    if let Some(scope) = scope {
        state.db.with_conn(|conn| {
            dao::rewrite_prompt_id(conn, scope, old_name.trim(), new_name.trim())?;
            Ok(())
        })?;
    }
    Ok(())
}

/// Delete a preset (idempotent).
#[tauri::command]
pub fn delete_prompt(name: String, target: Option<PromptTarget>) -> AppResult<()> {
    prompts::delete_prompt(target.unwrap_or_default(), &name)
}

/// Activate a preset: overwrite the live `~/.claude/CLAUDE.md` (backup first).
#[tauri::command]
pub fn activate_prompt(name: String, target: Option<PromptTarget>) -> AppResult<()> {
    prompts::activate_prompt(target.unwrap_or_default(), &name)
}

/// Read the current live `CLAUDE.md`, or `None` when absent.
#[tauri::command]
pub fn read_live_prompt(target: Option<PromptTarget>) -> AppResult<Option<LivePrompt>> {
    prompts::read_live_prompt(target.unwrap_or_default())
}

/// Import the live `CLAUDE.md` into the preset library under `name`.
#[tauri::command]
pub fn import_live_prompt(name: String, target: Option<PromptTarget>) -> AppResult<()> {
    prompts::import_live_prompt(target.unwrap_or_default(), &name)
}

File: src/services/prompts.ts
import { call } from "./ipc";
import type {
  LivePrompt,
  PromptDetail,
  PromptInfo,
  PromptTarget,
} from "@/types/backend";

export async function listPrompts(target: PromptTarget = "claude_code"): Promise<PromptInfo[]> {
  return call<PromptInfo[]>("list_prompts", { target });
}

export async function readPrompt(name: string, target: PromptTarget = "claude_code"): Promise<PromptDetail> {
  return call<PromptDetail>("read_prompt", { name, target });
}

export async function savePrompt(name: string, content: string, target: PromptTarget = "claude_code"): Promise<void> {
  return call<void>("save_prompt", { name, content, target });
}

export async function renamePrompt(
  oldName: string,
  newName: string,
  target: PromptTarget = "claude_code",
): Promise<void> {
  return call<void>("rename_prompt", { oldName, newName, target });
}

export async function deletePrompt(name: string, target: PromptTarget = "claude_code"): Promise<void> {
  return call<void>("delete_prompt", { name, target });
}

export async function activatePrompt(name: string, target: PromptTarget = "claude_code"): Promise<void> {
  return call<void>("activate_prompt", { name, target });
}

export async function readLivePrompt(target: PromptTarget = "claude_code"): Promise<LivePrompt | null> {
  return call<LivePrompt | null>("read_live_prompt", { target });
}

export async function importLivePrompt(name: string, target: PromptTarget = "claude_code"): Promise<void> {
  return call<void>("import_live_prompt", { name, target });
}

export async function listPiPromptTemplates(): Promise<string[]> {
  return call<string[]>("list_pi_prompt_templates", {});
}

export async function readPiPromptTemplate(name: string): Promise<string> {
  return call<string>("read_pi_prompt_template", { name });
}

export async function savePiPromptTemplate(name: string, content: string): Promise<void> {
  return call<void>("save_pi_prompt_template", { name, content });
}

export async function deletePiPromptTemplate(name: string): Promise<void> {
  return call<void>("delete_pi_prompt_template", { name });
}

export async function getWorkspacePiPrompt(workspaceDir: string): Promise<[string, string] | null> {
  return call<[string, string] | null>("get_workspace_pi_prompt", { workspaceDir });
}

export async function saveWorkspacePiPrompt(workspaceDir: string, fileName: string, content: string): Promise<void> {
  return call<void>("save_workspace_pi_prompt", { workspaceDir, fileName, content });
}

// ---- Skills -----------------------------------------------------------------
